### PR TITLE
📝 ♻️ Refactor structure of the docs and integrate legacy docs

### PR DESCRIPTION
**Pull request by patrikspiess**
As discussed ...

### CODE REVIEW

**Review by lucmurer — CHANGES_REQUESTED** [review]
Still two things are open I think:
- Remove unneeded indirection for ForitClientEMS
- FortiGate config check part should go to the user documentation

File: docs/source/developer/how_to/fortinet/forticlientems.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+.. _how_to_forticlientems:
+
+How To Use FortiClient EMS Devices
+==================================
+
+.. toctree::
+  :maxdepth: 1
+  
+  forticlientems/forticlientems_login
```
[the remark this answers is not in the document]
Comment (patrikspiess) [reply docs/source/developer/how_to/fortinet/forticlientems.rst]: It's better now

File: docs/source/developer/how_to/fortinet/fortigate.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.. Describes how to use FortiGate devices
+
+.. _how_to_fortigate:
+
+How To Use FortiGate Devices
+============================
+
```
[the remark this answers is not in the document]
Comment (patrikspiess) [reply docs/source/developer/how_to/fortinet/fortigate.rst:7]: Please review

File: docs/source/developer/how_to/fortinet/fortigate/fortigate_config_check.rst [hunk 3]
```diff
@@ -0,0 +1,203 @@
+.. Describes how to use the FortiGate configuration check
+
+.. _how_to_fortigate_config_check:
+
+The FortiGate Configuration Check
+=================================
+
+
+The FortiGate configuration check module let's you check the configuration of a FortiGate.
```
[the remark this answers is not in the document]
Comment (patrikspiess) [reply docs/source/developer/how_to/fortinet/fortigate/fortigate_config_check.rst]: Fixed

File: docs/source/developer/how_to/fortinet/fortigate/fortigate_config_check.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
+.. Describes how to use the FortiGate configuration check
+
+.. _how_to_fortigate_config_check:
+
+The FortiGate Configuration Check
```
[the remark this answers is not in the document]
Comment (patrikspiess) [reply docs/source/developer/how_to/fortinet/fortigate/fortigate_config_check.rst]: You're right. Moved to the user docs

**Review by lucmurer — APPROVED** [review]

**Pull request merged**